applications/spacewalk-bridge.md

```diff
@@ -0,0 +1,231 @@
+# W3F Grant Proposal
+
+- **Project Name:** Spacewalk: a Stellar bridge
+- **Team Name:** Pendulum
+- **Payment Address:** [0x41826C59a853969DA6B819130E1c32A9fd7c94ba](https://etherscan.io/address/0x41826C59a853969DA6B819130E1c32A9fd7c94ba#tokentxns) (DAI)
+- **[Level](https://github.com/w3f/Grants-Program/tree/master#level_slider-levels):** 2
+
+
+## Project Overview :page_facing_up:
+
+### Overview
+
+Spacewalk is a bridge between Substrate-based parachains and [Stellar](https://stellar.org) which enables asset transfers to and from Stellar. This grant application is for developing the *Spacewalk protocol and pallet*. The Spacewalk bridge is built by the team behind the [Pendulum](http://pendulumchain.org) network (an upcoming parachain that connects fiat tokens from across multiple blockchain ecosystems).
+
+### Project Details
+
+Spacewalk bridge is developed to be a decentralized and trustless bridge to Stellar. This bridge enables two main activities:
+
+- **Deposit**: bridge any Stellar asset to Substrate-based chains. The tokens of the source asset are transferred to some dedicated **bridge account** in Stellar where they are locked. The bridge then mints wrapped tokens in the target chain and transfers them to the recipient account.
+- **Withdrawal**: bridge wrapped tokens in Substrate-based chains to Stellar. First, the user instructs the bridge to burn wrapped tokens in the Substrate-based chain. Afterwards, the bridge unlocks tokens held in the bridge account in Stellar and transfers them to some target account.
+
+Stellar is not smart contract capable – therefore we follow the recommendation laid out in the [Polkadot documentation](https://wiki.polkadot.network/docs/learn-bridges#via-higher-order-protocols) and base our bridge design on XCLAIM. In XCLAIM, the bridge account is called a **vault**. XCLAIM is based on the following four core features:
+
+- Implement a chain relay for Stellar in the bridge pallet
+- Employ collateralization in order to ensure that the vault exhibits good behavior
+- Ensure that the economic value of the collateral exceeds the value of the vault
+- Enable a decentralized network of vaults
+
+Our bridge differs from XCLAIM in some details as follows:
+
+- Implementing a full chain relay for Stellar is out of scope of this web3 grant proposal as this requires an update to the Stellar consensus protocol. Instead the Spacewalk protocol assumes that a light Stellar node is co-located for every node of the Substrate-based chain and uses the Stellar nodes as an oracle.
+- We will employ a single vault instead of a network of vaults because according to Section V.E. of [Xclaim: Trustless, interoperable, cryptocurrency-backed assets](https://eprint.iacr.org/2018/643.pdf), the latter version leads to non-fungible wrapped tokens which is not a desirable condition.
```

Comment:
This is no longer how Interlay's bridge works - wrapped tokens can now be redeemed with any vault, not just the issuing one. Check out [their docs](https://docs.interlay.io/#/vault/overview). A permissionless vault market is preferable to a single (permissioned) vault.

Reply:
Already at the superficial level, simply consider censorability and single-point-of-failure downsides.